src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -205,6 +205,10 @@ protected function handleCircularReference($object)
     protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
     {
         if (!$this->classMetadataFactory) {
+            if (isset($context[static::ALLOW_EXTRA_ATTRIBUTES]) && !$context[static::ALLOW_EXTRA_ATTRIBUTES]) {
+                throw new LogicException('A class metadata factory must be provided in the constructor when setting ALLOW_EXTRA_ATTRIBUTES to false.');
```

Comment:
Actually should rather use the `ALLOW_EXTRA_ATTRIBUTE` const value here I think.

Reply:
Where is that constant defined? I don't see that anywhere in the project, at least on the 3.4 and 4.1 branches.

Reply:
I mean `static::ALLOW_EXTRA_ATTRIBUTES`. the one used in the `isset` above ^^

Reply:
Oh, I get it - the docs and so on don't use the class constant, but the actual lowercase string. I'll update.

Reply:
That's why I asked for using the const value. Here you're mentioning the constant in the error message :)